tests/desktop-e2e/scripts/test-cross-client-sync.sh

```
@@ -186,6 +194,82 @@ else
+  INNER_CONTENT=$(cat "$MP/$RENAMED_DIR/inner.txt" 2>/dev/null || echo "")
+  if [ "$INNER_CONTENT" = "$RENAME_INNER_CONTENT" ]; then
+    pass "FUSE mount picked up folder rename with intact content"
+  else
+    fail "FUSE mount renamed folder but inner content mismatch (got: '$INNER_CONTENT')"
+  fi
+else
+  fail "FUSE mount did not detect folder rename after 120s"
+fi
+
+# Cleanup renamed folder
+rm -rf "$MP/$RENAMED_DIR" 2>/dev/null || true
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)